src/lib/controller.ts

```diff
@@ -145,6 +160,170 @@ class FastAuthController {
   async signTransaction(params) {
     return this.signDelegateAction(params);
   }
+
+  // This call need to be called after new oidc token is generated
+  // https://github.com/near/mpc-recovery#claim-oidc-id-token-ownership
+  async claimOidcToken(oidcToken) {
+    const CLAIM_SALT = 3177899144 + 0;
+    const keypair = await this.getKey();
+    const signObj = getUserCredentialsFrpSignature({
+      salt: CLAIM_SALT,
+      oidcToken,
+      shouldHashToken: true,
+      keypair,
+    });
+
+    const data = {
+      oidc_token_hash: sha256(oidcToken),
+      frp_signature: Buffer.from(signObj.signature).toString('hex'),
+      frp_public_key: keypair.getPublicKey().toString(),
+    };
+
+    // https://github.com/near/mpc-recovery#claim-oidc-id-token-ownership
+    // TODO: replace newMpcRecoveryUrl to mpcRecovery when all endpoint is implemented
+    return fetch(`${network.fastAuth.newMpcRecoveryUrl}/claim_oidc`, {
+      method: 'POST',
+      mode: 'cors' as const,
+      body: JSON.stringify(data),
+      headers: new Headers({ 'Content-Type': 'application/json' }),
+    }).then(async (response) => {
+      if (!response.ok) {
+        throw new Error('Unable to claim OIDC token');
+      }
+      const res = await response.json();
+      return res.mpc_signature;
```

Comment:
To be sure that your token was claimed, you will need to check if this signature is correct using mpc_pk.
Check this code for reference: https://github.com/near/mpc-recovery/blob/develop/integration-tests/src/containers.rs#L838